src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1048,41 +1069,52 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
             }
         }
 
-        /// <summary>
-        /// Authorization check on mutation fields provided in a GraphQL Mutation request.
-        /// </summary>
-        /// <param name="context"></param>
-        /// <param name="parameters"></param>
-        /// <param name="entityName"></param>
-        /// <param name="mutationOperation"></param>
-        /// <exception cref="DataApiBuilderException"></exception>
+        /// <inheritdoc/>
         public void AuthorizeMutationFields(
+            string inputArgumentName,
             IMiddlewareContext context,
+            string clientRole,
             IDictionary<string, object?> parameters,
             string entityName,
             EntityActionOperation mutationOperation)
         {
-            string role = GetRoleOfGraphQLRequest(context);
+            if (mutationOperation is EntityActionOperation.Create)
```

Comment:
Looks like for create, we call AuthorizeEntityAndFieldsForMutation irrespective of point or multiple mutations, right?

Reply:
yes